megameklab/src/megameklab/ui/util/TabStateUtil.java

```diff
@@ -0,0 +1,172 @@
+        var dir = getTabStateDirectory();
+
+        var clean = new File(dir, TAB_STATE_CLEAN);
+        if (clean.exists()) {
+            if (!clean.delete()) {
+                throw new IOException("Could not delete " + clean);
+            }
+        }
+
+        FileUtils.cleanDirectory(dir);
+
+        Map<File, String> filenameAssociations = new LinkedHashMap<>();
+
+        for (var editor : editors) {
+            File unitFile;
+            if (editor.getEntity() instanceof Mek) {
+                unitFile = File.createTempFile("mml_unit_", ".mtf.tmp", dir);
+                try (
+                    var fos = new FileOutputStream(unitFile);
+                    var ps = new PrintStream(fos)
+                    ) {
+                    ps.println(((Mek) editor.getEntity()).getMtf());
+                } catch (Exception e) {
+                    continue;
+                }
+            } else {
+                unitFile = File.createTempFile("mml_unit_", ".blk.tmp", dir);
+                try {
+                    BLKFile.encode(unitFile.getPath(), editor.getEntity());
+                } catch (EntitySavingException e) {
+                    continue;
+                }
+            }
+
+            var fileName = editor.getFileName();
+            if (fileName == null || fileName.isBlank()) {
+                fileName = " ";
+            }
+
+            filenameAssociations.put(unitFile, fileName);
+        }
+
+        File filenameAssociationsFile = new File(dir, FILENAME_ASSOCIATIONS);
+        try (
+            var fos = new FileOutputStream(filenameAssociationsFile);
+            var ps = new PrintStream(fos)
+        ) {
+            for (var entry : filenameAssociations.entrySet()) {
+                ps.print(entry.getKey().getPath() + '\0' + entry.getValue() + '\0');
+            }
+        }
+
+        clean.createNewFile();
+    }
+
+    public static List<MegaMekLabMainUI> loadTabState() throws IOException {
+        var dir = getTabStateDirectory();
+        List<MegaMekLabMainUI> editors = new ArrayList<>();
+
+        var clean = new File(dir, TAB_STATE_CLEAN);
+        if (!clean.exists()) {
+            return editors;
+        }
+
+        var db = new File(dir, FILENAME_ASSOCIATIONS);
+        if (!db.exists()) {
+            return editors;
+        }
+
+        var parts = Files.readString(Paths.get(db.getAbsolutePath())).split(Pattern.quote("\0"));
+        for (int i = 0; i < parts.length; i += 2) {
+            var entityFile = new File(parts[i]);
+
+            var newFile = new File(entityFile.getAbsolutePath().replaceFirst("\\.tmp$", ""));
+            FileUtils.copyFile(entityFile, newFile);
+
+            var fileName = parts[i + 1];
+            if (fileName.isBlank()) {
+                fileName = "";
+            }
+
+            try {
+                Entity loadedUnit = new MekFileParser(newFile).getEntity();
+                newFile.delete();
```

Comment:
I'd argue that we shouldn't delete any files until we've completed all actions in the `try` block.